base/compiler/abstractinterpretation.jl

```diff
@@ -261,8 +295,10 @@ function abstract_call_method(method::Method, @nospecialize(sig), sparams::Simpl
                         parent_method2 = parent.src.method_for_inference_limit_heuristics # limit only if user token match
                         parent_method2 isa Method || (parent_method2 = nothing) # Union{Method, Nothing}
                         if (parent.cached || parent.limited) && parent.linfo.def === sv.linfo.def && sv_method2 === parent_method2
-                            topmost = infstate
-                            edgecycle = true
+                            if !argtypes_strictly_less_comples(get_argtypes(parent.result), get_argtypes(sv.result))
```

Comment:
`get_argtypes` is very slow, and thus is not permitted to be used. It's also only an approximation, so again, not permitted to be used.

Reply:
IUCC the result of actually computing `get_argtypes` should already be cached here, yeah? I don't think we ever construct an `InferenceResult` instance without immediately computing the argtypes. So it should basically just be a field access by this point. 

I did some refactoring in #28955 to make things like this less confusing - after that gets merged, `get_argtypes(parent.result)` here would just be `parent.result.argtypes`.